lib/rubygems/specification.rb

```diff
@@ -782,8 +782,18 @@ def self.default_stubs pattern
   private_class_method :default_stubs
 
   def self.installed_stubs dirs, pattern
-    map_stubs(dirs, pattern) do |path, base_dir, gems_dir|
-      Gem::StubSpecification.gemspec_stub(path, base_dir, gems_dir)
+    # if mingw, derive platform from file name (loaded_from)
```

Comment:
does this need to be hard-coded for `mingw`, or should it apply for any platform ?

Reply:
Looking over the code and `Gem::StubSpecification::StubLine`, I think the following would work, and would be much more generic:

```ruby
  def self.installed_stubs dirs, pattern
    map_stubs(dirs, pattern) do |path, base_dir, gems_dir|
      Gem::StubSpecification.gemspec_stub(path, base_dir, gems_dir)
    end.select { |s| Gem::Platform.match s.platform }
  end
```

Thoughts?  Also, should I add the test?

Reply:
That does look a lot better to me

Reply:
Ok.  Add the test?  Thanks, Greg